apiserver/testing/stub_network.go

```diff
@@ -486,6 +487,18 @@ func (sb *StubBacking) AdditionalSubnets() {
 	}
 }
 
+func (sb *StubBacking) SpaceByName(name string) (networkingcommon.BackingSpace, error) {
```

Comment:
As we discussed, these are to go from here.

Reply:
For example:
```diff
diff --git a/apiserver/facades/client/spaces/spaces_test.go b/apiserver/facades/client/spaces/spaces_test.go
index 5eb1c6421f..20edb9724d 100644
--- a/apiserver/facades/client/spaces/spaces_test.go
+++ b/apiserver/facades/client/spaces/spaces_test.go
@@ -259,6 +259,22 @@ func (s *SpaceTestMockSuite) expectMachines(ctrl *gomock.Controller, addresses s
 	s.mockBacking.EXPECT().AllMachines().Return(mockMachines, machErr)
 }
 
+type stubBacking struct {
+	*apiservertesting.StubBacking
+}
+
+func (sb *stubBacking) SpaceByName(name string) (networkingcommon.BackingSpace, error) {
+	panic("should not be called")
+}
+
+func (sb *stubBacking) AllEndpointBindings() ([]spaces.ApplicationEndpointBindingsShim, error) {
+	panic("should not be called")
+}
+
+func (sb *stubBacking) AllMachines() ([]spaces.Machine, error) {
+	panic("should not be called")
+}
+
 // This is the old testing suite
 type SpacesSuite struct {
 	coretesting.BaseSuite
@@ -303,7 +319,7 @@ func (s *SpacesSuite) SetUpTest(c *gc.C) {
 	s.blockChecker = mockBlockChecker{}
 	var err error
 	s.facade, err = spaces.NewAPIWithBacking(
-		apiservertesting.BackingInstance,
+		&stubBacking{apiservertesting.BackingInstance},
 		&s.blockChecker,
 		s.callContext,
 		s.resources, s.authorizer,
@@ -322,7 +338,7 @@ func (s *SpacesSuite) TearDownTest(c *gc.C) {
 func (s *SpacesSuite) TestNewAPIWithBacking(c *gc.C) {
 	// Clients are allowed.
 	facade, err := spaces.NewAPIWithBacking(
-		apiservertesting.BackingInstance,
+		&stubBacking{apiservertesting.BackingInstance},
 		&s.blockChecker,
 		s.callContext,
 		s.resources, s.authorizer,
@@ -336,7 +352,7 @@ func (s *SpacesSuite) TestNewAPIWithBacking(c *gc.C) {
 	agentAuthorizer := s.authorizer
 	agentAuthorizer.Tag = names.NewMachineTag("42")
 	facade, err = spaces.NewAPIWithBacking(
-		apiservertesting.BackingInstance,
+		&stubBacking{apiservertesting.BackingInstance},
 		&s.blockChecker,
 		context.NewCloudCallContext(),
 		s.resources,
@@ -749,7 +765,7 @@ func (s *SpacesSuite) TestReloadSpacesUserDenied(c *gc.C) {
 	agentAuthorizer := s.authorizer
 	agentAuthorizer.Tag = names.NewUserTag("regular")
 	facade, err := spaces.NewAPIWithBacking(
-		apiservertesting.BackingInstance,
+		&stubBacking{apiservertesting.BackingInstance},
 		&s.blockChecker,
 		context.NewCloudCallContext(),
 		s.resources, agentAuthorizer,
@@ -765,7 +781,7 @@ func (s *SpacesSuite) TestSuppportsSpacesModelConfigError(c *gc.C) {
 		errors.New("boom"), // Backing.ModelConfig()
 	)
 
-	err := spaces.SupportsSpaces(apiservertesting.BackingInstance, context.NewCloudCallContext())
+	err := spaces.SupportsSpaces(&stubBacking{apiservertesting.BackingInstance}, context.NewCloudCallContext())
 	c.Assert(err, gc.ErrorMatches, "getting environ: boom")
 }
 
@@ -776,7 +792,7 @@ func (s *SpacesSuite) TestSuppportsSpacesEnvironNewError(c *gc.C) {
 		errors.New("boom"), // environs.New()
 	)
 
-	err := spaces.SupportsSpaces(apiservertesting.BackingInstance, context.NewCloudCallContext())
+	err := spaces.SupportsSpaces(&stubBacking{apiservertesting.BackingInstance}, context.NewCloudCallContext())
 	c.Assert(err, gc.ErrorMatches, "getting environ: boom")
 }
 
@@ -788,7 +804,7 @@ func (s *SpacesSuite) TestSuppportsSpacesWithoutNetworking(c *gc.C) {
 		apiservertesting.WithoutSpaces,
 		apiservertesting.WithoutSubnets)
 
-	err := spaces.SupportsSpaces(apiservertesting.BackingInstance, context.NewCloudCallContext())
+	err := spaces.SupportsSpaces(&stubBacking{apiservertesting.BackingInstance}, context.NewCloudCallContext())
 	c.Assert(err, jc.Satisfies, errors.IsNotSupported)
 }
 
@@ -807,12 +823,12 @@ func (s *SpacesSuite) TestSuppportsSpacesWithoutSpaces(c *gc.C) {
 		errors.New("boom"), // Backing.supportsSpaces()
 	)
 
-	err := spaces.SupportsSpaces(apiservertesting.BackingInstance, context.NewCloudCallContext())
+	err := spaces.SupportsSpaces(&stubBacking{apiservertesting.BackingInstance}, context.NewCloudCallContext())
 	c.Assert(err, jc.Satisfies, errors.IsNotSupported)
 }
 
 func (s *SpacesSuite) TestSuppportsSpaces(c *gc.C) {
-	err := spaces.SupportsSpaces(apiservertesting.BackingInstance, context.NewCloudCallContext())
+	err := spaces.SupportsSpaces(&stubBacking{apiservertesting.BackingInstance}, context.NewCloudCallContext())
 	c.Assert(err, jc.ErrorIsNil)
 }
 
diff --git a/apiserver/testing/stub_network.go b/apiserver/testing/stub_network.go
index f06005c5d7..e73468390d 100644
--- a/apiserver/testing/stub_network.go
+++ b/apiserver/testing/stub_network.go
@@ -16,7 +16,6 @@ import (
 	"gopkg.in/juju/names.v3"
 
 	"github.com/juju/juju/apiserver/common/networkingcommon"
-	"github.com/juju/juju/apiserver/facades/client/spaces"
 	"github.com/juju/juju/core/instance"
 	"github.com/juju/juju/core/life"
 	"github.com/juju/juju/core/network"
@@ -487,18 +486,6 @@ func (sb *StubBacking) AdditionalSubnets() {
 	}
 }
 
-func (sb *StubBacking) SpaceByName(name string) (networkingcommon.BackingSpace, error) {
-	panic("implement me")
-}
-
-func (sb *StubBacking) AllEndpointBindings() ([]spaces.ApplicationEndpointBindingsShim, error) {
-	panic("implement me")
-}
-
-func (sb *StubBacking) AllMachines() ([]spaces.Machine, error) {
-	panic("implement me")
-}
-
 func (sb *StubBacking) ModelConfig() (*config.Config, error) {
 	sb.MethodCall(sb, "ModelConfig")
 	if err := sb.NextErr(); err != nil {
```

Reply:
thanks!